edx_solutions_api_integration/users/views.py

```diff
@@ -636,12 +636,17 @@ def post(self, request, user_id):  # pylint: disable=R0915
         if existing_user_profile:
             if first_name and last_name:
                 existing_user_profile.name = u'{} {}'.format(first_name, last_name)
-            city = request.data.get('city')
-            if city:
-                existing_user_profile.city = city
-            country = request.data.get('country')
-            if country:
-                existing_user_profile.country = country
+
+            existing_user_profile.city = request.data.get('city')
+            existing_user_profile.country = request.data.get('country')
+            existing_user_profile.gender = request.data.get('gender')
+
+            # Empty title is also allowed
+            existing_user_profile.title = request.data.get('title', existing_user_profile.title)
```

Comment:
I think this appraoch should be applied to `city`, `country`, `gender` and unit tests should be updated to test scenarios where if `city`, `country`, `gender` and `title` is missing it should not be updated.